src/io.c

```diff
@@ -1,4478 +1,4647 @@
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_ACCEL:		/* acceleration */
+#ifdef OUTPUT_ACCELERATION
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k = 0; k < 3; k++) {fp[k] = All.cf_a2inv * P[pindex].GravAccel[k];}
+#ifdef PMGRID
+                    for(k = 0; k < 3; k++) {fp[k] += All.cf_a2inv * P[pindex].GravPM[k];}
+#endif
+                    if(P[pindex].Type == 0) {for(k = 0; k < 3; k++) {fp[k] += SphP[pindex].HydroAccel[k];}}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_DTENTR:		/* rate of change of internal energy */
+#ifdef OUTPUT_CHANGEOFENERGY
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DtInternalEnergy;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_DELAYTIME:
+#ifdef GALSF_SUBGRID_WINDS
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DelayTime;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_TSTP:		/* timestep  */
+#ifdef OUTPUT_TIMESTEP
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = GET_PARTICLE_TIMESTEP_IN_PHYSICAL(pindex);
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_BFLD:		/* magnetic field  */
+#ifdef MAGNETIC
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = (MyOutputFloat) (Get_Gas_BField(pindex,k) * All.cf_a2inv * gizmo2gauss);}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_VDIV:		/* Divergence of Vel */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = P[pindex].Particle_DivVel;
+                    n++;
+                }
+            break;
+
+        case IO_VORT:		/* Vorticity */
+#if defined(TURB_DRIVING) || defined(OUTPUT_VORTICITY)
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = SphP[pindex].Vorticity[k];}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_SLUG_STATE_RNG:  /* It is an 128 bit integer. I split it into 2 64 bit integers for easier output */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    x = P[pindex].slug_cluster_state.rngStateAtBirth;

```

Comment:
You are missing a type declaration here. To avoid these kinds of errors, I would suggest that you try to compile the code and see if you get any error messages.